llvm/test/CodeGen/AMDGPU/while-break.ll

```diff
@@ -152,4 +152,87 @@ end:
   ret float %r
 }
 
+; Two chains of phi network that have the same value from %if block.
+define amdgpu_ps < 2 x float> @while_break_two_chains_of_phi(float %v, i32 %x, i32 %y, i32 %z, ptr addrspace(1) %p) #0 {
```

Comment:
This should be copied/moved to test/Transforms/StructurizeCFG to demonstrate the effect in LLVM IR. That was very useful for me to see the optimization working. At least under StructurizeCFG, it does not need to be specific to AMDGPU.

Reply:
see #102118